### PR TITLE
Upgrade Nette

### PR DESCRIPTION
**Pull request by VBoss**
Coding standard
- upper cased null/false/true
- global function fqn
- use strict types
- return types
- typed method arguments when possible

Travis
- Unified travis.yml
- Added PHPStan to travis
- Added CodingStandard to travis
- Removed unused nette libraries

Composer
- Added PHPStan (dev)
- Added Kdyby/CodingStandard (dev)
- Moved PHPCoveralls to composer (dev)
- Added Typo3/ClassAliasLoader (dev)
- Moved PHPParallelLint to composer (dev)
- bump Nette/Caching to 3.0
- bump Nette/Deprecated to 3.0 (dev)
- Autoload set to PSR4

Exception split to own files
- Old namespaces aliased by ClassAliasLoder

Nette 3.0 compatibility
- changed removed setClass -> setType
- setType has only one argument, arguments are passed in setArguments method
- Removed unused register method
- PHP71+

CodingStandard + PHPStan + Other
- Removed `Kdyby\Redis\IRedisDriver::class` and `Kdyby\Redis\IRedisDriverOld::class` because they had outdated methods and docblocks, needed to be updated with redis changes. Also they had same methods as `\Redis` and as interfaces they needed to have same type hints as implemented classes wich depends on installed version on client server. All this meant 30+ CS and PHPStan skipped rules and no added value. So use default `\Redis` hinting and if needed add some undocumented stuff add it in docblok of `Kdyby\Redis\Driver\PhpRedisDriver::class` since it is named driver it is appropriate place.
- Added types to method arguments where possible, skipped where problematic
- Added return types to methods where possible, skipped where problematic
- Specified types in docblock where it can not be declared specificaly
- `class RedisLuaJournal extends RedisJournal implements Nette\Caching\Storages\IJournal` -> `class RedisLuaJournal extends \Kdyby\Redis\RedisJournal`
- Exctracted classes from `AbstractRedisTestCase.php` to own files PSR4 style
- Tons of PHPStan fixes
- Minor modification to test to run in 2.4 env `Nette\Http\Request` has one less parameter

TODO
- Bump Tester to 2.2 and rework stress tests to work with new tester

### CODE REVIEW

File: tests/KdybyTests/bootstrap.php [hunk 1]
```diff
@@ -8,31 +10,37 @@
...
 }
 
+// because of old nette/tester
+error_reporting(E_ALL & ~(E_DEPRECATED | E_WARNING));
+
 // configure environment
 Tester\Environment::setup();
-class_alias('Tester\Assert', 'Assert');
-date_default_timezone_set('Europe/Prague');
+\class_alias('Tester\Assert', 'Assert');
```
Comment (VBoss) [inline tests/KdybyTests/bootstrap.php]: can this be in typo3 aliases?

File: tests/php.ini-unix [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [php]
 session.gc_probability=0
+error_reporting=E_ALL & ~( E_WARNING | E_DEPRECATED )
```
Comment (VBoss) [inline tests/php.ini-unix]: Nette/tester 1.7 related

**Comment by elcheco** [comment]
I'd like to ask how does it look like with this Nette 3 upgrade? On Nette COnference was mentioned that's almost done and it's quite a while and no progress... Any help needed or what's the blocker here?

**Comment by JakubBarta** [comment]
Hi @fprochazka , can you pls have a look at this PR? all checks are passing

**Comment by fprochazka** [comment]
cc @VBoss

**Comment by VBoss** [comment]
on it

**Comment by JakubBarta** [comment]
hi @VBoss , any update? can I help you with something?

**Comment by VBoss** [comment]
I am not completly sure tests cover all important use cases. It would be great if someone could test this PR on their codebase and let me know if there is some problem. We are slowly testing this with colleagues but our project is far from done.

As alternative I am thinking of some sandbox/playground package for Kdyby where everything (dependencies, loading, DI, saving) can be tested.

**Comment by VBoss** [comment]
Redis test implementation is here https://github.com/VBoss/sandbox/blob/master/app/presenters/RedisPresenter.php
Composer - was OK with Nette 3
Configuring - with no problem
Usability - there could be bit of information in documentation, about injecting cache to Nette DI so you dont use storage and journal directly, but this is specific with level of experience of developer who is implementing redis.

Overall everythink looks fine. So lets do this.

**Comment by KacerCZ** [comment]
@VBoss It seems that nette/reflection package was accidenally moved from "require-dev" to "require" section. Reflection is used only in tests.

**Pull request merged**